tooling/storybook-addon/src/feature/direction/useDirection.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from "react"
+import { addons } from "@storybook/addons"
+import { EVENTS } from "../../constants"
+
+/**
+ * Sync the layout direction to the html element
+ */
+export const useDirection = (initialDirection: "ltr" | "rtl" = "ltr") => {
+  const [direction, setDirection] = React.useState(
+    initialDirection.toLowerCase(),
+  )
+
+  React.useEffect(() => {
+    document.documentElement.dir = direction
```

Comment:
This would toggle the entire storybook layout to RTL and I think that might be unexpected. I'd prefer if we change the direction for just the storybook iframe preview area.

```jsx
const $el = document.getElementById("storybook-preview-iframe")
      const iframe = $el as HTMLIFrameElement
      iframe.contentDocument!.dir = state.direction
```

Reply:
The `useDirection` hook is rendered inside of the story, so it does not toggle the direction on the outer html element, just the one in the iframe. This screenshot shows both elements:
<img width="499" alt="image" src="https://user-images.githubusercontent.com/16899513/166148489-709e9853-7011-4a40-8724-8a0eef15834d.png">